lib/activerecord_spanner_adapter/information_schema.rb

```diff
@@ -213,9 +217,7 @@ def execute_query sql, params = {}
       params = params.transform_values { |v| quote v }
       sql = format sql, params
 
-      @mutex.synchronize do
-        @connection.snapshot(sql, strong: true).rows
-      end
+      @connection.execute_query(sql).rows
```

Comment:
Creating a snapshot for each query is not necessary, and this will save a round-trip to the server.